doc/CONFIGURATION.md

```diff
@@ -505,6 +505,32 @@ const node = await Libp2p.create({
 })
 ```
 
+#### Configuring PeerStore
+
+PeerStore persistence is disabled in libp2p by default. You can enable and configure it as follows. Aside from enabled being `false` by default, it will need an implementation of a [datastore](https://github.com/ipfs/interface-datastore).
+
+```js
+const Libp2p = require('libp2p')
+const TCP = require('libp2p-tcp')
+const MPLEX = require('libp2p-mplex')
+const SECIO = require('libp2p-secio')
+
+const LevelStore = require('datastore-level')
+
+const node = await Libp2p.create({
+  modules: {
+    transport: [TCP],
+    streamMuxer: [MPLEX],
+    connEncryption: [SECIO]
+  },
+  datastore: new LevelStore('path/to/store'),
+  peerStore: {
+    persistence: true,
+    threshold: 5
+  }
```

Comment:
We need to make sure to document the options somewhere so it's clear what each one does

Reply:
I agree with this. Where do you think we should do it?

We are also not doing it for the dialer, metrics, connectionManager... We should probably add all of them in the `API.md`, after the create? I should probably do a follow up PR with all of them?

Reply:
I think at least having comments in the Configuration.md examples for each section that show the defaults would be helpful. We have it in some of the other sections, just not everywhere. A followup PR works for me

Reply:
I added comments as you recommended for the PeerStore now. But, I will make sure to add proper docs as a follow up PR to all the components